src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Models/FilterScheme.cs

```diff
@@ -99,8 +99,9 @@ private void OnConditionItemsChanged()
             SubscribeToEvents();
         }
 
-        protected override void OnDeserialized()
+        protected async override void OnDeserialized()
         {
+            await this.EnsureIntegrityAsync();
```

Comment:
I double checked, I see I made a mistake by marking the non-async obsolete. Please remove the ObsoleteEx() from the non-async method as well.